### ISSUE TITLE
best_model에 dev dataset 넣어서 결과 보는 코드

### ISSUE THREAD

**Issue by hoooolllly** [boostcampaitech5/level2_klue-nlp-07#7]
```python
from transformers import AutoTokenizer, AutoConfig, AutoModelForSequenceClassification, Trainer, TrainingArguments
from torch.utils.data import DataLoader
from load_data import *
import pandas as pd
import torch
import torch.nn.functional as F

import pickle as pickle
import numpy as np
import argparse
from tqdm import tqdm

def inference(model, tokenized_sent, device):
  """
    test dataset을 DataLoader로 만들어 준 후,
    batch_size로 나눠 model이 예측 합니다.
  """
  dataloader = DataLoader(tokenized_sent, batch_size=32, shuffle=False)
  model.eval()
  output_pred = []
  output_prob = []
  for i, data in enumerate(tqdm(dataloader)):
    with torch.no_grad():
      outputs = model(
          input_ids=data['input_ids'].to(device),
          attention_mask=data['attention_mask'].to(device),
          token_type_ids=data['token_type_ids'].to(device)
          )
    logits = outputs[0]
    prob = F.softmax(logits, dim=-1).detach().cpu().numpy()
    logits = logits.detach().cpu().numpy()
    result = np.argmax(logits, axis=-1)

    output_pred.append(result)
    output_prob.append(prob)
  
  return np.concatenate(output_pred).tolist(), np.concatenate(output_prob, axis=0).tolist()

def num_to_label(label):
  """
    숫자로 되어 있던 class를 원본 문자열 라벨로 변환 합니다.
  """
  origin_label = []
  with open('dict_num_to_label.pkl', 'rb') as f:
    dict_num_to_label = pickle.load(f)
  for v in label:
    origin_label.append(dict_num_to_label[v])
  
  return origin_label

def load_dev_dataset(dev_dir, tokenizer):
  dev_dataset = load_data(dev_dir)
  dev_label = [100 for _ in range(len(dev_dataset))]
  # tokenizing dataset
  tokenized_dev = tokenized_dataset(dev_dataset, tokenizer)
  return dev_dataset['id'], tokenized_dev, dev_label

def is_correct(pred_answer,answer):
  tmp = []
  for pred,ans in zip(pred_answer,answer):
    if pred == ans:
      tmp.append(0)
    else:
      tmp.append(1)
  return tmp

def main(args):
  """
    주어진 dataset csv 파일과 같은 형태일 경우 inference 가능한 코드입니다.
  """
  device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')
  # load tokenizer
  Tokenizer_NAME = "klue/roberta-large"
  tokenizer = AutoTokenizer.from_pretrained(Tokenizer_NAME)

  ## load my model
  MODEL_NAME = args.model_dir # model dir.
  model = AutoModelForSequenceClassification.from_pretrained(args.model_dir)
  model.parameters
  model.to(device)


  dev_dir = "../../dataset/dev/dev.csv"
  dev_dataset = load_data(dev_dir)
  dev_real_label = list(dev_dataset['label'])
  dev_sentence_ = list(dev_dataset['sentence'])
  dev_subj_ = list(dev_dataset['subject_entity'])
  dev_obj_ = list(dev_dataset['object_entity'])
    
  dev_id, dev_dataset, dev_label = load_dev_dataset(dev_dir, tokenizer)
  Re_dev_dataset = RE_Dataset(dev_dataset ,dev_label)
  
  
  ## predict answer
  pred_answer, output_prob = inference(model, Re_dev_dataset, device) # model에서 class 추론
  pred_answer = num_to_label(pred_answer) # 숫자로 된 class를 원래 문자열 라벨로 변환.

  a = is_correct(pred_answer,dev_real_label)

  output = pd.DataFrame({'id':dev_id,'sentence':dev_sentence_,'subject_entity':dev_subj_,'object_entity':dev_obj_,'label':dev_real_label,'pred_label':pred_answer,'is_correct':a,'probs':output_prob,})
  label_sort = list(output['label'].value_counts().keys())
  print('(틀린개수/전체개수)')
  for i in label_sort:
    total = output['label'].value_counts()[i]
    try:
        wrong = output[output['is_correct'] == 1]['label'].value_counts()[i]
        ratio = round(wrong / total,3)
        print(i,':','(',wrong,'/',total,')', '오답률 : ', ratio)
    except:
        print(i,':','(',0,'/',total,')', '오답률 : ', 0)
        
  output.to_csv('./prediction_ver2/dev_submission.csv', index=False) # 최종적으로 완성된 예측한 라벨 csv 파일 형태로 저장.
  #### 필수!! ##############################################
  print('---- Finish! ----')
if __name__ == '__main__':
  parser = argparse.ArgumentParser()
  
  # model dir
  parser.add_argument('--model_dir', type=str, default="./best_model_ver2")
  args = parser.parse_args()
  print(args)
  main(args)
``` 
inference.py 살짝 수정한 코드입니다. 경로만 바꾸면 됩니다.
각 라벨마다 맞은개수, 틀린개수 출력되고 csv파일은 아래의 열을 가집니다.
id | sentence | subject_entity | object_entity | label | pred_label | is_correct | probs
